src/Raven.Pal/src/posix/mapping.c

```diff
@@ -267,3 +277,44 @@ rvn_discard_virtual_memory(void* address, int64_t size, int32_t* detailed_error_
         *detailed_error_code = errno;
     return rc;
 }
+
+EXPORT int32_t
+rvn_remap(void *base_address, void **new_address, void *handle, int64_t size, int32_t flags, int64_t offset, int32_t *detailed_error_code)
+{
+    int32_t rc = rvn_unmap(MMOPTIONS_NONE, base_address, ALLOCATION_GRANULARITY, detailed_error_code);
```

Comment:
This is *not* safe to do.
When we remap, we create a _new_ mapping. The old one is still there and valid.
We will be disposing that _separately_. If you don't handle this, you'll have crashes when read transactions that hold a reference to the old mapping are trying to read it.